File: mall-admin/src/api/seckill/seckillOrder.ts
import { get } from '../request'
import type { PageResult } from '../request'

export interface SeckillOrder {
  id?: number
  orderSn?: string
  sessionId?: number
  sessionName?: string
  skuId?: number
  skuName?: string
  memberId?: number
  username?: string
  seckillPrice: number
  quantity: number
  totalAmount: number
  status: number // 1: 待支付, 2: 已支付, 3: 已取消
  payTime?: string
  createTime?: string
}

export interface SeckillOrderParams {
  key?: string
  sessionId?: number
  status?: number
}

// 获取秒杀订单列表
export const getSeckillOrderList = (params?: SeckillOrderParams) => {
  return get<PageResult<SeckillOrder>>('/admin/seckill/order/list', params)
}

// 根据 ID 获取秒杀订单
export const getSeckillOrderById = (id: number) => {
  return get<SeckillOrder>(`/admin/seckill/order/${id}`)
}

File: mall-seckill/src/main/java/com/hj/mall/seckill/entity/SeckillOrder.java
package com.hj.mall.seckill.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 秒杀订单实体
 */
@Data
@TableName("sms_seckill_order")
public class SeckillOrder {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 会员ID
     */
    private Long memberId;

    /**
     * 秒杀场次ID
     */
    private Long sessionId;

    /**
     * 商品SKU ID
     */
    private Long skuId;

    /**
     * 订单ID
     */
    private Long orderId;

    /**
     * 订单编号
     */
    private String orderSn;

    /**
     * 秒杀价格
     */
    private BigDecimal seckillPrice;

    /**
     * 购买数量
     */
    private Integer quantity;

    /**
     * 状态：0->无效；1->有效
     */
    private Integer status;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;
}
